xcc/settings.py

```diff
@@ -87,15 +87,45 @@ class Config:  # pylint: disable=missing-class-docstring
         env_file = get_path_to_env_file()
         env_prefix = get_name_of_env_var()
 
+    def _sanity_check(self, key: str, val: str) -> None:
+        """
+        Check for conditions that make saving the env_file
+        dangerous to the user.
+
+        Parameters
+        ----------
+        key : str
+            env file key
+        val : str
+            env file value
+
+        Raises
+        ------
+        Exception
+            When the value should not be saved to the env_file.
+
+        Returns
+        -------
+        None
+            Has no effect if all is safe.
```

Comment:
We should format this docstring in the same style as the rest of the functions and classes (similar to [PennyLane](https://pennylane.readthedocs.io/en/stable/development/guide/documentation.html)):
```suggestion
        Args:
            key (str): .env file key
            val (str): .env file value

        Raises:
            Exception: if the value should not be saved to the .env file
```
Note that there is no need to document the return value if the function does not return a value!